defaults/main.yml

```diff
@@ -1,6 +1,8 @@
 ---
 # docker-engine is the default package name
-docker_pkg_name: docker-engine
+docker_pkg_version: 17.05.0
+docker_pkg_edition: ce
+docker_pkg_name: docker-engine={{ docker_pkg_version }}~{{ docker_pkg_edition }}-0~{{ansible_lsb.id|lower }}-{{ ansible_lsb.codename|lower }}
```

Comment:
Let's get rid of the "magic" and move as much of the right hand side of this into a single variable.

If the "right hand variable" is set to `latest`, use that to replicate the `update_docker_package` behaviour and remove that variable.

Reply:
Would we want to set it to latest? Wouldn't that kinda defeat the point of pinning?